### DISCUSSION TITLE
Other Emwin page 500 server error

### DISCUSSION THREAD

**Discussion by stonegrift** [JVital2013/vitality-goes#83] [Q&A] [answered]
I performed the install per the instructions on the main page. After letting the system run a couple of days I had enough data to allow the Local Settings page to populate.  Different issue the Current Weather page populates correctly, but if I go to the other emwin page to browse other data, the sub tabs show up for a moment then it updates to  a 500 server error and doesn't even show the select data.

**Comment by stonegrift** [comment]
a bit more information. I turned on debug in the config.ini file.  The other emwin page reported two errors. 

Warning:  Undefined array key "format" in /var/www/html/functions.php on line 115

Warning:  Undefined array key "truncate" in /var/www/html/functions.php on line 116

Followed by an incredibly long line listing  what looks in passing to be every videoPath  and title from the scriptconfig.ini  I've attached what I got.  when I got this information. it pretty much crashed the server. I had to edit the config to turn debug back off before the page would load again.
[Errors.txt](https://github.com/user-attachments/files/16117899/Errors.txt)

**Comment by stonegrift** [comment]
In trying to figure this out I watched the otheremwin page load.  Before the 500 error it actually shows all the tiles. And if I'm quick enough I can open them before the error blanks the page.  They even populate (at least the add data tile does.  The other say loading)  So I think it's something that might be in the function.php or in my file storage.  I am not sure  on this since the current weather page populates WITH the correct emwin data with no errors.  Still pondering what could be going on.  I even tried reinstalling Vitality while leaving everything in the config folder alone.  I half wonder if there's been an update to some back end product 'apache2', 'php' that is forcing an error.  I've not looked into the dataHandler.php simply because trying to follow through calls to multiple pages begins to make my eyes cross to keep things straight.

**Comment by stonegrift** [comment]
I noticed a couple of things and I"m going to try them next.  One I live in an area where the RWR and the NWS office are the same place. one will populate on the local config and the other won't if  part of it is populated already.  Such as mine the RWR office is set to SEW WA.  and the NWS Office won't even give the option of choosing Washington.  It's like the state left the union!  Second thing I'm going to try.  I took a look at the Raspberry Pi imager and discovered I can deliberately install a legacy version.  I can't help thinking that there is some back end stuff that was done to make the OS 'better' but because I'm so naive when it comes to Linux. And that's shooting me in the foot. Joournalctl comes to mind.

**Comment by stonegrift** [comment, marked as answer]
Well, after starting over, installing Bullseye rather than Bookworm this time. After making sure goestools was actually downloading emwin messages, I let it run a while  After an hour or so I installed vitality-goes again.  Initially, the choices didn't completely populate  I think this would be another option for a website scrape of office names and states OR at the very least make it very clear that there needs to be at least a day's worth of emwin data to allow Vitality to gather the names it needs because without that, it's very frustrating to go through setting things up and not everything works right out of the box.

**Reply by BigBayWx** [reply]
I had the exact same issue.  No matter what I tried with Bookworm it wouldn't work but when I started over with Bullseye and did a complete reinstall it worked fine right out of the box.
